### ISSUE TITLE
Sitemap does not generated

### ISSUE THREAD

**Issue by dmitrys99** [redline6561/coleslaw#84]
When I include `(sitemap)` to plugins setting nothing happens after call to `(coleslaw:main ...)` while other files generated successfully.
Seems that sitemap plugin does not even get called.
Latest master branch, sbcl 1.2.13, FreeBSD.

**Comment by redline6561** [comment]
Thanks for the bug report! I'll get a fix for this written this weekend.

**Comment by PuercoPop** [comment]
It makes sense that it doesn't get called.

The sitemap plugin 'hooks' itself by defining a publish method to a class that is neither a subclass of content or index, so [`#'compile-blog`](https://github.com/redline6561/coleslaw/blob/master/src/coleslaw.lisp#L35) will never call it. However that appears to be a deliberate decision https://github.com/redline6561/coleslaw/commit/78f1fc9fa05c4134e8fe9cd2069350da8ebf6458

Although the problem is that sitemap needs to be called last, I can't quite grasp what was the intended solution.

The least hacky solution I can think of, but hacky nonetheless, is to define an `*after-publish-hook*` in `#'compile-blog` and let the sitemap plugin hook to it.

**Comment by dmitrys99** [comment]
Maybe it will be good to have some other hooks exported? Before publish, for example?
Or maybe make plugin have method `execute` to be called when appropriate?

**Comment by redline6561** [comment]
Indeed, the previous "intended solution" was clearly not thought through on my part. This is a big part of why my next big Coleslaw task during my vacation the next 2 weeks will be to get a solid test suite written and Travis CI set up.

I think from a design perspective, turning compile-blog (and maybe also deploy) into a Generic Function that takes a *Config* object is an interesting idea. Then there could be an `:around` method defined on compile-blog to do the directory handling stuff and run `(call-next-method)` inside the `with-staging-directory` stuff. Anyway, I will have a look at it as soon as I can. This week at work is particularly crazy so it will be a few days.

**Comment by ehouse** [comment]
Have there been any updates on this? I noticed it still isn't getting generated on master.
